swim/labels_test.go

```diff
@@ -0,0 +1,298 @@
+		"key1":       "new value for 1",
+		"__private1": "private value1",
+	}
+
+	fixtureLabelsRoom = map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+		"key3": "value3",
+		"key4": "value4",
+	}
+
+	fixtureLabelsRoomWithPrivate = map[string]string{
+		"key1":       "value1",
+		"key2":       "value2",
+		"key3":       "value3",
+		"key4":       "value4",
+		"__private1": "private value1",
+	}
+
+	fixtureExactLabels = map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+		"key3": "value3",
+		"key4": "value4",
+		"key5": "value5",
+	}
+
+	fixtureExactLabelsWithPrivate = map[string]string{
+		"key1":       "value1",
+		"key2":       "value2",
+		"key3":       "value3",
+		"key4":       "value4",
+		"key5":       "value5",
+		"__private1": "private value1",
+	}
+
+	fixtureTooBigLabels = map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+		"key3": "value3",
+		"key4": "value4",
+		"key5": "value5",
+		"key6": "value6",
+	}
+
+	fixtureTooBigLabelsWithPrivate = map[string]string{
+		"key1":       "value1",
+		"key2":       "value2",
+		"key3":       "value3",
+		"key4":       "value4",
+		"key5":       "value5",
+		"key6":       "value6",
+		"__private1": "private value1",
+	}
+)
+
+func TestMergeLabelOptions(t *testing.T) {
+	empty := LabelOptions{}
+	merged := mergeLabelOptions(empty, DefaultLabelOptions)
+	assert.Equal(t, DefaultLabelOptions, merged, "expected an empty LabelOptions struct to equal the DefaultLabelOptions when merged")
+
+	onlyKeySize := LabelOptions{KeySize: 1}
+	merged = mergeLabelOptions(onlyKeySize, DefaultLabelOptions)
+	assert.Equal(t, 1, merged.KeySize, "expected the KeySize to be set to the value set in the optional LabelOptions")
+	assert.Equal(t, DefaultLabelOptions.ValueSize, merged.ValueSize, "expected the ValueSize to be taken from the default options")
+	assert.Equal(t, DefaultLabelOptions.Count, merged.Count, "expected the Count to be taken from the default options")
+
+	onlyValueSize := LabelOptions{ValueSize: 1}
+	merged = mergeLabelOptions(onlyValueSize, DefaultLabelOptions)
+	assert.Equal(t, 1, merged.ValueSize, "expected the ValueSize to be set to the value set in the optional LabelOptions")
+	assert.Equal(t, DefaultLabelOptions.KeySize, merged.KeySize, "expected the KeySize to be taken from the default options")
+	assert.Equal(t, DefaultLabelOptions.Count, merged.Count, "expected the Count to be taken from the default options")
+
+	onlyCount := LabelOptions{Count: 1}
+	merged = mergeLabelOptions(onlyCount, DefaultLabelOptions)
+	assert.Equal(t, 1, merged.Count, "expected the Count to be set to the value set in the optional LabelOptions")
+	assert.Equal(t, DefaultLabelOptions.ValueSize, merged.ValueSize, "expected the ValueSize to be taken from the default options")
+	assert.Equal(t, DefaultLabelOptions.KeySize, merged.KeySize, "expected the KeySize to be taken from the default options")
+}
+
+// This test relies on the default settings set to:
```

Comment:
Might be "safer" to explicitly set them so we don't need to change tests when the defaults are changed.